### PR TITLE
Adding Telegraf v4 support

### PR DESCRIPTION
**Pull request by Viiprogrammer**
In Telegraf v4 `updateSubTypes` was removed in [#1219](https://github.com/telegraf/telegraf/pull/1219), and it throws an error on line when pushing subtype in removed context field.

I added checking for `updateSubTypes` exists before pushing and add example for v4 version :smiley:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,4 +20,24 @@ bot.on('media_group', (ctx) => {
...
   return ctx.reply(`total: ${ctx.mediaGroup.length}`)
 })
-```
\ No newline at end of file
+```
+
+## Telegraf v4
+```js
+const { Telegraf } = require('telegraf')
+const mediaGroup = require('.')
```
Comment (Loskir) [inline README.md]:
```suggestion
const mediaGroup = require('telegraf-media-group')
```

File: README.md [hunk 2]
```diff
@@ -18,4 +20,24 @@ bot.on('media_group', (ctx) => {
...
+
+const bot = new Telegraf(process.env.BOT_TOKEN)
+
+bot.use(mediaGroup())
+
+bot.on('message', async (ctx) => {
+  // ctx.mediaGroup — an array of album messages (including the last one)
+  for (const message of ctx.mediaGroup) {
```
Comment (Loskir) [inline README.md:36]: Isn't it `undefined` if this message is not part of media group?

File: package.json [hunk 3]
```diff
@@ -6,6 +6,6 @@
   "author": "Loskir",
   "license": "MIT",
   "dependencies": {
-    "telegraf": "^3.36.0"
+    "telegraf": ">= 3.36.0"
```
Comment (Loskir) [inline package.json]: I suggest using something like `^3.36.0||^4.0.0`

Comment (Loskir) [reply package.json]: Honestly it should be a peer dependency instead

**Review by Loskir — CHANGES_REQUESTED** [review]

**Comment by Loskir** [comment]
Thank you for your contribution 👍

File: package.json (hunk 3 above)
Comment (Viiprogrammer) [reply package.json]: And what about yarn.lock, is it not required in this case?

File: README.md (hunk 2 above)
Comment (Viiprogrammer) [reply README.md:36]: I thought this was just an example, and the check could be omitted, but I found a better solution for version 4, which gives more compatibility and does not require checks.

File: package.json [hunk 4]
```diff
@@ -5,7 +5,7 @@
   "repository": "https://github.com/Loskir/telegraf-media-group.git",
   "author": "Loskir",
   "license": "MIT",
-  "dependencies": {
-    "telegraf": "^3.36.0"
+  "peerDependencies": {
+    "telegraf": "^3.39.0"
```
Comment (Loskir) [inline package.json]:
```suggestion
    "telegraf": "^3.36.0 || ^4.0"
```

I think it shouldn't be the latest version, but rather the first supported one. I'm too lazy to check so let's leave it as is. Also added v4.

Comment (Viiprogrammer) [reply package.json]: I added v4, but it looks like when installing modules for tests, I missed it and only v3 got into the commit, I should be more careful ....

**Comment by Loskir** [comment]
Version 1.2 with support for Telegraf v4 is now released 👏

**Pull request merged**